### ISSUE TITLE
Deadlock when using TransitDecryptAsync from synchronous code

### ISSUE THREAD

**Issue by rune-berge** [rajanadar/VaultSharp#39]
Hi
I have an asp.net web api where I want to add some encryption using Vault Transit functionality. 

Here is the wrapper method I've written
```
public string TransitDecrypt(string encryptedValue)
{
    var value = CreateTransitClient().TransitDecryptAsync(TransitKeyName, encryptedValue).Result;

    return Encoding.UTF8.GetString(Convert.FromBase64String(value.Data.PlainText));
}
```

When calling this from synchronous api code the Result method never returns, and the api eventually times out. I tested the same code in a command line application and it works fine. I assume this is a deadlock situation. Rewriting the entire api to support aync all the way up is not something I want to do...

The vault client is initialized with continueAsyncTasksOnCapturedContext: false

Then I downloaded the source code for VaultSharp and debugged it. Then I got as far as the following line:
https://github.com/rajanadar/VaultSharp/blob/380ce0c7e06437198932129669c53d79da8251a4/src/VaultSharp/VaultClient.cs#L2099

I added ConfigureAwait(_continueAsyncTasksOnCapturedContext) to this await, and also to the other await in this method (see below), and suddenly everything works
https://github.com/rajanadar/VaultSharp/blob/380ce0c7e06437198932129669c53d79da8251a4/src/VaultSharp/VaultClient.cs#L2107

Is there a reason configureAwait is not used here, or is this a bug?

**Comment by rune-berge** [comment]
So, this is how the method looks after my modifications:

```
       private async Task<TResponse> MakeVaultApiRequest<TResponse>(string resourcePath, HttpMethod httpMethod, object requestData = null, bool rawResponse = false, Func<int, string, TResponse> customProcessor = null, string wrapTimeToLive = null) where TResponse : class
        {
            var headers = new Dictionary<string, string>();

            if (_lazyVaultToken != null)
            {
                headers.Add(VaultTokenHeaderKey, await _lazyVaultToken.Value.ConfigureAwait(_continueAsyncTasksOnCapturedContext));
            }

            if (wrapTimeToLive != null)
            {
                headers.Add(VaultWrapTimeToLiveHeaderKey, wrapTimeToLive);
            }

            return await _dataAccessManager.MakeRequestAsync<TResponse>(resourcePath, httpMethod, requestData, headers, rawResponse, customProcessor).ConfigureAwait(_continueAsyncTasksOnCapturedContext);
        }
```

**Comment by niralatuladhar** [comment]
Even I encountered same issue. Basically, we cannot call from the synchronous api.  Can you please put this fix and release the new version.

**Comment by rajanadar** [comment]
@rune-berge @niralatuladhar looks like a bug. Thanks @rune-berge for this.

I'll try to get out a specific release with just this fix, by end of week.

**Comment by niralatuladhar** [comment]
Thank you @rajanadar

**Comment by rajanadar** [comment]
Will be solved as part of #48

**Comment by rajanadar** [comment]
@rune-berge @niralatuladhar 

The latest version of VaultSharp is out. It solves the problem by relaying the async-context flag cleanly
Do check it out.
